src/Stack/Build/Execute.hs

```diff
@@ -229,10 +229,13 @@ executePlan menv bopts baseConfigOpts locals sourceMap plan = do
             , eeGlobalDB = globalDB
             }
 
-    unless (Map.null $ planInstallExes plan) $ do
+    when (installExesPlanFlag $ planInstallExes plan) $ do
+        -- Also install executables
         snapBin <- (</> bindirSuffix) `liftM` installationRootDeps
         localBin <- (</> bindirSuffix) `liftM` installationRootLocal
-        destDir <- asks $ configLocalBin . getConfig
```

Comment:
Okay, I've been staring at this tonight for far too long and am finally making sense of it. So, I can keep the hardcode "$HOME/.local/bin" in the config file and then call `loadConfigWitithOpts` overriding the `destDir`. Then I can clean up the code in the middle to remove the floating install path? I've pushed the WIP, but it's my first time now making sense of the local config Monoid.